test/dynamo/test_repros.py

```diff
@@ -3284,6 +3284,72 @@ def fn(x):
         ref = opt_fn(x)
         self.assertEqual(ref, res)
 
+    def test_sum_dtype(self):
```

Comment:
Actually this is not a Dynamo bug, it's a decomposition bug, it's better we can move the unit test to ```torch/testing/_internal/common_methods_invocations.py```, you can just add a similar item like this:
https://github.com/pytorch/pytorch/blob/dfa64fddeb6b58c60a59096202b192ea09f6ed40/torch/testing/_internal/common_methods_invocations.py#L20214-L20219

Reply:
Thanks for the review. I see that above this there is: https://github.com/pytorch/pytorch/blob/dfa64fddeb6b58c60a59096202b192ea09f6ed40/torch/testing/_internal/common_methods_invocations.py#L20209-L20213

I just want to confirm that you'd like me to add a `PythonRefInfo` in addition to this `ReductionPythonRefInfo`, is that the case?

Reply:
That's a good question, I'm not pretty sure. @ngimel can you suggest which is the right way?

Reply:
If I understand the test correctly, the ops test that maybe could have caught this issue is this one: 
https://github.com/pytorch/pytorch/blob/dfa64fddeb6b58c60a59096202b192ea09f6ed40/test/test_ops.py#L786-L802

But I don't think it actually tests this behavior at all since it always provides a dtype in all the cases, and the bug fixed by this PR requires that the dtype not be provided. It also doesn't exercise the case where the provided dtype is an integral type which would be exercised in case 4 but that only runs if the dtype is a floating point type.